server/controllers/list.js

```diff
@@ -605,70 +668,114 @@ const changeToMember = (req, resp) => {
+        const { status, message } = err;
+        return resp.status(status).send({ message });
+      }
+
+      resp.status(400).send({ message: 'List data not found' });
     });
 };
 
-const addMember = (req, resp) => {
+const addViewer = (req, resp) => {
   const {
     user: { _id: currentUserId }
   } = req;
   const { id: listId } = req.params;
   const { email } = req.body;
+  let list;
+  let user;
+  let cohortMembers;
 
-  List.findOne({ _id: listId, $or: [{ ownerIds: currentUserId }] })
-    .populate('cohortId', 'memberIds ownerIds')
+  List.findOne({
+    _id: listId,
+    $or: [{ ownerIds: currentUserId }]
+  })
+    .populate('cohortId', 'ownerIds memberIds')
     .exec()
     .then(doc => {
-      if (doc) {
-        const { cohortId: cohort, ownerIds } = doc;
+      list = doc;
 
-        return User.findOne({ email })
-          .exec()
-          .then(doc => {
-            const { _id, avatarUrl, displayName } = doc;
-            return { _id, avatarUrl, cohort, displayName, ownerIds };
-          })
-          .catch(() => {
-            throw new BadRequestException('User data not found.');
-          });
+      return User.findOne({ email }).exec();
+    })
+    .then(userData => {
+      if (!userData) {
+        throw new BadRequestException(`There is no user of email: ${email}`);
+      }
+
+      const { _id: newMemberId } = userData;
+      const { cohortId: cohort, viewersIds } = list;
+      const userNotExists = !checkIfArrayContainsUserId(
+        viewersIds,
+        newMemberId
+      );
+
+      user = userData;
+
+      if (userNotExists) {
+        list.viewersIds.push(newMemberId);
       }
 
-      return resp
-        .status(400)
-        .send({ message: "You don't have permission to add new member" });
+      if (cohort) {
+        const { memberIds } = cohort;
+        cohortMembers = memberIds;
+      }
+
+      return list.save();
     })
-    .then(data => {
-      const {
-        _id: newMemberId,
-        cohort,
-        displayName,
-        avatarUrl,
-        ownerIds
-      } = data;
-
-      List.findOneAndUpdate(
-        { _id: listId, memberIds: { $nin: [newMemberId] } },
-        { $push: { memberIds: newMemberId } }
-      )
-        .exec()
-        .then(doc => {
-          if (doc) {
-            const data = { avatarUrl, displayName, newMemberId };
-            const dataToSend = responseWithListMember(data, ownerIds, cohort);
-
-            return resp.status(200).json(dataToSend);
-          }
-
-          return resp
-            .status(400)
-            .send({ message: 'User is already a member.' });
-        })
-        .catch(() => {
-          throw new BadRequestException('List data not found.');
-        });
+    .then(() => {
+      let userToSend;
+
+      if (cohortMembers) {
+        userToSend = responseWithListMember(user, cohortMembers);
+      } else {
+        userToSend = responseWithListMember(user, []);
+      }
```

Comment:
could be just:
`const userToSend = responseWithListMember(user, cohortMembers ? cohortMembers : []);`

Reply:
can't use this because of es lint rule 
Unnecessary use of conditional expression for default assignment

but I have found another solution